### PR TITLE
Added basic WebApp for opening/closing door with LDAP logins Added door_lib which abstracts away door handling

### PR DESCRIPTION
**Pull request by cfstras**

### CODE REVIEW

File: door_lib.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    line = s.readline()
+    logging.debug("line: {}".format(line))
+    first = line.split(" ")[0]
+    try:
+        adc = int(first)
+    except ValueError:
+        msg = "ValueError: '%s' is not a valid door state" % first
+        logging.error(msg)
+        print(msg)
+        return False, first
```
Comment (malled2002) [inline door_lib.py:47]: y not throwing an exception? otherwise this could indicate an incorrect door state

File: web.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+@app.route('/logout')
+def logout():
+    session.clear()
+    return redirect(url_for('index'))
+
+@app.route('/door', methods=['POST'])
+def door():
+    if request.form.get('action', 'closedoor') == 'opendoor':
+
+        if 'uid' not in session:
```
Comment (malled2002) [inline web.py:60]: without ssl everyone on the network can hijack a session cookie

**Review by malled2002 — APPROVED** [review]
+1

File: door_lib.py (hunk 1 above)
Comment (cfstras) [reply door_lib.py:47]: if we ensure everybody handles the exception,/
 sure.

File: web.py (hunk 2 above)
Comment (cfstras) [reply web.py:60]: better than sending the pw with every request... but yeah we need ssl for this

**Pull request merged**